buildroot use https

ahaoboy opened the pull request:
Using https can avoid the following errors in ci

```
2025-03-05 05:59:24 fv-az1390-389 root[2627] ERROR running command git fetch --force --progress --update-head-ok -- git@github.com:lynx-family/buildroot.git +refs/heads/*:refs/remotes/origin/* in /home/runner/.habitat_cache/git/buildroot.git/95ee6ed180392706ece98afd1d1e087a, original output:
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

```

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/lynx-family/primjs?pullRequest=3) <br/>All committers have signed the CLA.

jianliang00 commented:
Thanks for your contribution. The changes look good, but we need some modifications in the CI workflows. Let me amend some changes to your patch

jianliang00 commented:
Would you mind to add following patch to your pull request, then I can start the landing process

```
commit 2c5f4db37a1d8bc8e04538bf23eb4f111e9a4729
Author: ahaoboy <504595380@qq.com>
Date:   Wed Mar 5 14:29:59 2025 +0800

    [Infra] Use https instead of SSH to download public source code.
    
    Using the HTTPS protocol to download code can avoid permission issues caused by incorrect SSH private key configuration.

diff --git a/.github/workflows/android-release.yml b/.github/workflows/android-release.yml
index 579a94b..26e2ed1 100644
--- a/.github/workflows/android-release.yml
+++ b/.github/workflows/android-release.yml
@@ -16,12 +16,6 @@ jobs:
     - name: Download Source
       uses: actions/checkout@v4.2.2
 
-    - name: Install SSH Key
-      uses: kielabokkie/ssh-key-and-known-hosts-action@v1
-      with:
-        ssh-private-key: ${{ secrets.PRIVATE_SSH_KEY }}
-        ssh-host: github.com
-
     - name: Set up JDK 11
       uses: actions/setup-java@v1
       with:
diff --git a/.github/workflows/build-aar.yml b/.github/workflows/build-aar.yml
index 5ad0dd2..012ab1e 100644
--- a/.github/workflows/build-aar.yml
+++ b/.github/workflows/build-aar.yml
@@ -15,11 +15,6 @@ jobs:
     steps:
     - name: Checkout repository
       uses: actions/checkout@v2
-    - name: Install SSH Key
-      uses: kielabokkie/ssh-key-and-known-hosts-action@v1
-      with:
-        ssh-private-key: ${{ secrets.PRIVATE_SSH_KEY }}
-        ssh-host: github.com
     - name: Set up JDK 11
       uses: actions/setup-java@v1
       with:
diff --git a/.github/workflows/ci.yml b/.github/workflows/ci.yml
index d85ac8a..51c3d68 100644
--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -20,11 +20,6 @@ jobs:
     steps:
       - name: Download Source
         uses: actions/checkout@v4.2.2
-      - name: Install SSH Key
-        uses: kielabokkie/ssh-key-and-known-hosts-action@v1
-        with:
-          ssh-private-key: ${{ secrets.PRIVATE_SSH_KEY }}
-          ssh-host: github.com
       - name: Bundle Install
         run: |-
           SDKROOT=/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk bundle install --path .bundle
@@ -37,11 +32,6 @@ jobs:
     steps:
       - name: Download Source
         uses: actions/checkout@v4.2.2
-      - name: Install SSH Key
-        uses: kielabokkie/ssh-key-and-known-hosts-action@v1
-        with:
-          ssh-private-key: ${{ secrets.PRIVATE_SSH_KEY }}
-          ssh-host: github.com
       - name: Set up Python 3.9
         uses: actions/setup-python@v5
         with:
@@ -63,11 +53,6 @@ jobs:
     steps:
       - name: Download Source
         uses: actions/checkout@v4.2.2
-      - name: Install SSH Key
-        uses: kielabokkie/ssh-key-and-known-hosts-action@v1
-        with:
-          ssh-private-key: ${{ secrets.PRIVATE_SSH_KEY }}
-          ssh-host: github.com
       - name: Install deps
         run: |
           source tools/envsetup.sh
diff --git a/.github/workflows/publish-pod.yml b/.github/workflows/publish-pod.yml
index d4c631c..b9f3b16 100644
--- a/.github/workflows/publish-pod.yml
+++ b/.github/workflows/publish-pod.yml
@@ -11,11 +11,6 @@ jobs:
     steps:
       - name: Download Source
         uses: actions/checkout@v4.2.2
-      - name: Install SSH Key
-        uses: kielabokkie/ssh-key-and-known-hosts-action@v1
-        with:
-          ssh-private-key: ${{ secrets.PRIVATE_SSH_KEY }}
-          ssh-host: github.com
       - name: Bundle Install
         run: |-
           SDKROOT=/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk bundle install --path .bundle
diff --git a/DEPS b/DEPS
index 5f0d484..6a438b8 100644
--- a/DEPS
+++ b/DEPS
@@ -12,7 +12,7 @@ system = platform.system().lower()
 deps = {
     "build": {
         "type": "git",
-        "url": "git@github.com:lynx-family/buildroot.git",
+        "url": "https://github.com/lynx-family/buildroot.git",
         "commit": "b74a2ad3759ed710e67426eb4ce8e559405ed63f",
         "ignore_in_git": True,
         "condition": system in ["linux", "darwin", "windows"],
diff --git a/PrimJS.podspec b/PrimJS.podspec
index b99c812..66ca9bf 100644
--- a/PrimJS.podspec
+++ b/PrimJS.podspec
@@ -28,7 +28,7 @@ Pod::Spec.new do |s|
   s.license = "MIT"
   s.author = { "pandazyp" => "2823543594@qq.com" }
 
-  s.source = { :git => "git@github.com:lynx-family/primjs.git", :tag => s.version.to_s }
+  s.source = { :git => "https://github.com/lynx-family/primjs.git", :tag => s.version.to_s }
 
   s.compiler_flags = "-Wall", "-Wno-shorten-64-to-32", "-Os"
   s.ios.deployment_target = "9.0"

```

jianliang00 commented:
> ```
> commit 2c5f4db37a1d8bc8e04538bf23eb4f111e9a4729
> Author: ahaoboy <504595380@qq.com>
> Date:   Wed Mar 5 14:29:59 2025 +0800
> 
>     [Infra] Use https instead of SSH to download public source code.
>     
>     Using the HTTPS protocol to download code can avoid permission issues caused by incorrect SSH private key configuration.
> 
> diff --git a/.github/workflows/android-release.yml b/.github/workflows/android-release.yml
> index 579a94b..26e2ed1 100644
> --- a/.github/workflows/android-release.yml
> +++ b/.github/workflows/android-release.yml
> @@ -16,12 +16,6 @@ jobs:
>      - name: Download Source
>        uses: actions/checkout@v4.2.2
>  
> -    - name: Install SSH Key
> -      uses: kielabokkie/ssh-key-and-known-hosts-action@v1
> -      with:
> -        ssh-private-key: ${{ secrets.PRIVATE_SSH_KEY }}
> -        ssh-host: github.com
> -
>      - name: Set up JDK 11
>        uses: actions/setup-java@v1
>        with:
> diff --git a/.github/workflows/build-aar.yml b/.github/workflows/build-aar.yml
> index 5ad0dd2..012ab1e 100644
> --- a/.github/workflows/build-aar.yml
> +++ b/.github/workflows/build-aar.yml
> @@ -15,11 +15,6 @@ jobs:
>      steps:
>      - name: Checkout repository
>        uses: actions/checkout@v2
> -    - name: Install SSH Key
> -      uses: kielabokkie/ssh-key-and-known-hosts-action@v1
> -      with:
> -        ssh-private-key: ${{ secrets.PRIVATE_SSH_KEY }}
> -        ssh-host: github.com
>      - name: Set up JDK 11
>        uses: actions/setup-java@v1
>        with:
> diff --git a/.github/workflows/ci.yml b/.github/workflows/ci.yml
> index d85ac8a..51c3d68 100644
> --- a/.github/workflows/ci.yml
> +++ b/.github/workflows/ci.yml
> @@ -20,11 +20,6 @@ jobs:
>      steps:
>        - name: Download Source
>          uses: actions/checkout@v4.2.2
> -      - name: Install SSH Key
> -        uses: kielabokkie/ssh-key-and-known-hosts-action@v1
> -        with:
> -          ssh-private-key: ${{ secrets.PRIVATE_SSH_KEY }}
> -          ssh-host: github.com
>        - name: Bundle Install
>          run: |-
>            SDKROOT=/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk bundle install --path .bundle
> @@ -37,11 +32,6 @@ jobs:
>      steps:
>        - name: Download Source
>          uses: actions/checkout@v4.2.2
> -      - name: Install SSH Key
> -        uses: kielabokkie/ssh-key-and-known-hosts-action@v1
> -        with:
> -          ssh-private-key: ${{ secrets.PRIVATE_SSH_KEY }}
> -          ssh-host: github.com
>        - name: Set up Python 3.9
>          uses: actions/setup-python@v5
>          with:
> @@ -63,11 +53,6 @@ jobs:
>      steps:
>        - name: Download Source
>          uses: actions/checkout@v4.2.2
> -      - name: Install SSH Key
> -        uses: kielabokkie/ssh-key-and-known-hosts-action@v1
> -        with:
> -          ssh-private-key: ${{ secrets.PRIVATE_SSH_KEY }}
> -          ssh-host: github.com
>        - name: Install deps
>          run: |
>            source tools/envsetup.sh
> diff --git a/.github/workflows/publish-pod.yml b/.github/workflows/publish-pod.yml
> index d4c631c..b9f3b16 100644
> --- a/.github/workflows/publish-pod.yml
> +++ b/.github/workflows/publish-pod.yml
> @@ -11,11 +11,6 @@ jobs:
>      steps:
>        - name: Download Source
>          uses: actions/checkout@v4.2.2
> -      - name: Install SSH Key
> -        uses: kielabokkie/ssh-key-and-known-hosts-action@v1
> -        with:
> -          ssh-private-key: ${{ secrets.PRIVATE_SSH_KEY }}
> -          ssh-host: github.com
>        - name: Bundle Install
>          run: |-
>            SDKROOT=/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk bundle install --path .bundle
> diff --git a/DEPS b/DEPS
> index 5f0d484..6a438b8 100644
> --- a/DEPS
> +++ b/DEPS
> @@ -12,7 +12,7 @@ system = platform.system().lower()
>  deps = {
>      "build": {
>          "type": "git",
> -        "url": "git@github.com:lynx-family/buildroot.git",
> +        "url": "https://github.com/lynx-family/buildroot.git",
>          "commit": "b74a2ad3759ed710e67426eb4ce8e559405ed63f",
>          "ignore_in_git": True,
>          "condition": system in ["linux", "darwin", "windows"],
> diff --git a/PrimJS.podspec b/PrimJS.podspec
> index b99c812..66ca9bf 100644
> --- a/PrimJS.podspec
> +++ b/PrimJS.podspec
> @@ -28,7 +28,7 @@ Pod::Spec.new do |s|
>    s.license = "MIT"
>    s.author = { "pandazyp" => "2823543594@qq.com" }
>  
> -  s.source = { :git => "git@github.com:lynx-family/primjs.git", :tag => s.version.to_s }
> +  s.source = { :git => "https://github.com/lynx-family/primjs.git", :tag => s.version.to_s }
>  
>    s.compiler_flags = "-Wall", "-Wno-shorten-64-to-32", "-Os"
>    s.ios.deployment_target = "9.0"
> ```

ahaoboy committed:
[Infra] Use https instead of SSH to download public source code.

    Using the HTTPS protocol to download code can avoid permission issues caused by incorrect SSH private key configuration.

jianliang00 commented:
/land
